### PR TITLE
Grinch Murder ability is no longer instant

### PR DESCRIPTION
**Pull request by mc-oofert**
[BALANCE]

## About the PR 
Changes grinches murder ability to an action bar. Both of them need to stand (or the victim on the floor) still. About half as short as a changeling absorb.

https://user-images.githubusercontent.com/70376633/147760207-5617c8ce-6388-466a-99d1-6966ae741c43.mp4
(note: slightly old vid, dropping the heart replaced with flatline)

## Why's this needed? 
Getting instakilled and spaced 2 minutes into the shift isnt fun. Also murder ability is op.
This makes the victim capable of resisting such unfortunate events.
(Of course they can just stun and shrink their heart)
## Changelog

```changelog
(u)Jimmyl
(+)Grinch instakill ability is no longer instant
```

### CODE REVIEW

File: code/modules/antagonists/grinch/abilities/murder.dm [hunk 1]
```diff
@@ -1,6 +1,71 @@
...
+		..()
+
+		var/mob/ownerMob = owner
+		if(owner && ownerMob && target && get_dist(owner, target) <= 1)
+			boutput(ownerMob, "<span class='notice'>You have successfully induced flatline!</span>")
+			ownerMob.visible_message("<span class='alert'><b>[ownerMob] shrinks [target]'s heart down two sizes too small!</b></span>")
+			playsound(target.loc, 'sound/impact_sounds/Flesh_Tear_1.ogg', 75, 1, -1)
+			target.add_fingerprint(ownerMob) // Why not leave some forensic evidence?
+			target.contract_disease(/datum/ailment/malady/flatline, null, null, 1) // path, name, strain, bypass resist
+			//target.organHolder.drop_organ("heart") // a little trollin
```
Comment (ZeWaka) [inline code/modules/antagonists/grinch/abilities/murder.dm:59]: del

File: code/modules/antagonists/grinch/abilities/murder.dm [hunk 2]
```diff
@@ -1,6 +1,71 @@
+/datum/action/bar/private/icon/grinchFlatline
+	duration = 120
```
Comment (ZeWaka) [inline code/modules/antagonists/grinch/abilities/murder.dm:2]: whats 120

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/grinch/abilities/murder.dm (hunk 2 above)
Comment (mc-oofert) [reply code/modules/antagonists/grinch/abilities/murder.dm:2]: no idea

Comment (ZeWaka) [reply code/modules/antagonists/grinch/abilities/murder.dm:2]: Then why put it there?

Comment (mc-oofert) [reply code/modules/antagonists/grinch/abilities/murder.dm:2]: affects duration and i dont know what its measured in but its a good length

Comment (ZeWaka) [reply code/modules/antagonists/grinch/abilities/murder.dm:2]: you need to notate it with a time define then
https://hackmd.io/@goonstation/docs/%2F%40goonstation%2Fcode#Time-Defines

File: code/modules/antagonists/grinch/abilities/murder.dm [hunk 3]
```diff
@@ -1,11 +1,75 @@
...
 /datum/targetable/grinch/instakill
 	name = "Murder"
-	desc = "Induces instant cardiac arrest in a target."
+	desc = "Induces cardiac arrest in a target. You and your target must hold still for this."
 	targeted = 1
 	target_anything = 0
 	target_nodamage_check = 1
 	max_range = 1
-	cooldown = 4800
+	cooldown = 2400
```
Comment (Zonespace27) [inline code/modules/antagonists/grinch/abilities/murder.dm:72]:
```suggestion
	cooldown = 4 MINUTES
```

**Comment by ThatFiveGuys** [comment]
Its important to note that the current murder ability has a whopping 8 minutes of cooldown, this means generally youre going to be using this ability defensively as by using it offensively you could easily be wasting it in critical situations where it would be needed for your survival. I feel like the primary mistake of this pr is treating a defensive ability as an offensive one. With this pr, the only way youd be able to use this ability is by either standing next to someone for 15 seconds, or by having someone stunned for 15 seconds. The only way youd be able to use this ability with these changes is by either not being in a battle or by already winning the battle (because if youre able to stand still for 15 seconds with someone stunned, youve already won the battle). 

While switching this ability from a defensive one to an offensive one isnt inherently bad, this pr goes around this a bad way. Using this to finish off a stunned victim would be pretty unviable, if you have someone stunned for 15 seconds you can easily kill them by just hitting them. Giving them cardiac arrest and 0 brute (and only being able to do so every 4 minutes) is far worse then giving someone 150 brute and cardiac arrest, and to do the later all you need is a decent melee weapon while the former you need to stand still for 15 seconds and use this ability. The only way you'd be viably able to use this ability now is by standing next to someone, and dying because you stood next to someone for 15 seconds with no warning on your end isnt particularly fun or interesting for the victim (nor is it those things for the antag either). 

I feel like the current murder ability is fine balance wise, however that does not mean it is fun. The 8 minute cooldown paired with this being really the only ability as grinch for actually doing damage (unless youre a chemnerd), means honestly as is grinch is overall one of the weaker antags. Making the death more interesting for the victim would not be a bad thing however, something like making it behave similar to arcfiends jolt ability which gives the victim ample times to call for help may be a good way to take this ability.

**Comment by mc-oofert** [comment]
> Its important to note that the current murder ability has a whopping 8 minutes of cooldown, this means generally youre going to be using this ability defensively as by using it offensively you could easily be wasting it in critical situations where it would be needed for your survival. I feel like the primary mistake of this pr is treating a defensive ability as an offensive one. With this pr, the only way youd be able to use this ability is by either standing next to someone for 15 seconds, or by having someone stunned for 15 seconds. The only way youd be able to use this ability with these changes is by either not being in a battle or by already winning the battle (because if youre able to stand still for 15 seconds with someone stunned, youve already won the battle).
> 
> While switching this ability from a defensive one to an offensive one isnt inherently bad, this pr goes around this a bad way. Using this to finish off a stunned victim would be pretty unviable, if you have someone stunned for 15 seconds you can easily kill them by just hitting them. Giving them cardiac arrest and 0 brute (and only being able to do so every 4 minutes) is far worse then giving someone 150 brute and cardiac arrest, and to do the later all you need is a decent melee weapon while the former you need to stand still for 15 seconds and use this ability. The only way you'd be viably able to use this ability now is by standing next to someone, and dying because you stood next to someone for 15 seconds with no warning on your end isnt particularly fun or interesting for the victim (nor is it those things for the antag either).
> 
> I feel like the current murder ability is fine balance wise, however that does not mean it is fun. The 8 minute cooldown paired with this being really the only ability as grinch for actually doing damage (unless youre a chemnerd), means honestly as is grinch is overall one of the weaker antags. Making the death more interesting for the victim would not be a bad thing however, something like making it behave similar to arcfiends jolt ability which gives the victim ample times to call for help may be a good way to take this ability.

me when i immediately get rushed by some random grinch 9 minutes in and die instantly
it is definently an offensive ability since nobody uses it for defense,
and usually theres like 1 person attempting to actively kill a grinch so basically nobody gets to kill them because they get instakilled as soon as they get into melee ranged

also it has visible text messages and the victim and anyone else can see the "KILL" action bar

**Comment by TDHooligan** [comment]
> Its important to note that the current murder ability has a whopping 8 minutes of cooldown, this means generally youre going to be using this ability defensively as by using it offensively you could easily be wasting it in critical situations where it would be needed for your survival. I feel like the primary mistake of this pr is treating a defensive ability as an offensive one. 

or alternatively you can walk up to a guy you hate, kill and space them instantly, sit in a disposal chute for 8 minutes then do it again. abstracting it as 'defensive' or 'offensive' is about how -you- percieve it and does not affect the ability for another person to walk up to another and make them dead with nothing in between.

as for the changes, 4 minutes' cooldown seems reasonable but a 15-second channel time is 50% of the maximum stun time that exists, you might have a lot of grinches having a hard time using the skill, even on targets in deep crit (who are basically dead anyways) - that or you'll see many grinches with flashes and stun batons, for better or worse.

**Comment by mc-oofert** [comment]
> > Its important to note that the current murder ability has a whopping 8 minutes of cooldown, this means generally youre going to be using this ability defensively as by using it offensively you could easily be wasting it in critical situations where it would be needed for your survival. I feel like the primary mistake of this pr is treating a defensive ability as an offensive one.
> 
> or alternatively you can walk up to a guy you hate, kill and space them instantly, sit in a disposal chute for 8 minutes then do it again. abstracting it as 'defensive' or 'offensive' is about how -you- percieve it and does not affect the ability for another person to walk up to a player and make them dead with nothing in between for no reason.
> 
> as for the changes, 4 minutes' cooldown seems reasonable but a 15-second channel time is 50% of the maximum stun time that exists, you might have a lot of grinches having a hard time using the skill, even on targets in deep crit (who are basically dead anyways) - that or you'll see many grinches with flashes and stun batons, for better or worse.

hmm what if i made it ranged except getting out of sight cancels it

**Comment by TDHooligan** [comment]
> hmm what if i made it ranged except getting out of sight cancels it

i'd say it should be worth giving it like 5~8 seconds channel time, making it ranged just means it's a death sentence if you're locked in a room with said grinch but worthless if you have an exit

**Pull request closed without merging**